src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LoggerExtensions.cs

```diff
@@ -25,7 +25,10 @@ public static class LoggerExtensions
         /// <example>logger.LogDebug(0, exception, "Error while processing request from {Address}", address)</example>
         public static void LogDebug(this ILogger logger, EventId eventId, Exception? exception, string? message, params object?[] args)
         {
-            logger.Log(LogLevel.Debug, eventId, exception, message, args);
+            if (logger.IsEnabled(LogLevel.Debug))
```

Comment:
Which allocations should be saved by this pattern?

The `logger.IsEnabled`-check should be done before the call to `LogDebug` in user code.
Thus the allocation of `args` (the array) and potentially the creation of `message` can be saved. These objects are already existent when reaching the check you introduced here.

Or even better use [LoggerMessage.Define](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggermessage.define?view=dotnet-plat-ext-6.0) where that behavior can be set / use the (new) logging source generator.